### PR TITLE
sched/wdog: Revert wd_cancel semantics

### PR DESCRIPTION
**Pull request by Fix-Point**
## Summary
In the original implementation, if wdog is not ACTIVE, -EINVAL will be returned. The rp2040 i2c driver relies on this semantics to correctly release the semaphore.

## Impact
Revert the wd_cancel semantics.

## Testing
Tested on Raspberry Pi Pico.

### CODE REVIEW

**Comment by nuttxpr** [comment]
[**\[Experimental Bot, please feedback here\]**](https://github.com/search?q=repo%3Aapache%2Fnuttx+13552&type=issues)



**No, this PR does not meet the NuttX requirements.**

**Missing Information:**

* **Summary:**
    * Lacks a clear explanation of why the change is necessary. What problem does reverting the `wd_cancel` semantics solve? 
    * Needs to describe what functional part of the code is being changed (e.g., watchdog driver, I2C driver interaction).
    * Requires a detailed explanation of how the change works. Simply stating "Revert the wd_cancel semantics" is not sufficient. What was the old behavior, what is the new behavior, and how does it affect the interaction between the watchdog and I2C drivers?
    * Missing issue references if applicable.
* **Impact:**
    * Unclear if an existing feature is being changed or if this is a bug fix. 
    * The impact on users, builds, hardware, documentation, security, and compatibility is not addressed.
* **Testing:**
    * Lacks specifics about the build host used for testing (OS, CPU, compiler version).
    * "Tested on Raspberry Pi Pico" is insufficient. Specify the architecture (ARM), board configuration, and NuttX version.
    * No testing logs are provided. Include logs showing the issue before the change and the corrected behavior after the change.

**Recommendations:**

1. **Expand the Summary:** Provide a thorough explanation of the problem, the solution, and the affected code areas.
2. **Address all Impact Points:** Analyze and describe the potential impact of this change on all aspects mentioned in the requirements.
3. **Provide Detailed Testing Information:** Specify your build host, target environment, and include relevant testing logs demonstrating the issue and the fix.

By providing the missing information, you can ensure your PR meets the NuttX requirements and can be effectively reviewed and merged.

**Review by xiaoxiang781216 — APPROVED** [review]

File: sched/wdog/wd_cancel.c [hunk 1]
```diff
@@ -105,27 +105,24 @@ int wd_cancel_irq(FAR struct wdog_s *wdog)
 
   /* Make sure that the watchdog is still active. */
 
-  if (WDOG_ISACTIVE(wdog))
-    {
-      bool head = list_is_head(&g_wdactivelist, &wdog->node);
+  bool head = list_is_head(&g_wdactivelist, &wdog->node);
```
Comment (GUIDINGLI) [inline sched/wdog/wd_cancel.c:108]: move before line 94

**Review by linguini1 — APPROVED** [review]
This resolves the issue I was encountering with I2C. Thank you for the work!

**Review by acassis — APPROVED** [review]

File: sched/wdog/wd_cancel.c (hunk 1 above)
Comment (xiaoxiang781216) [reply sched/wdog/wd_cancel.c:108]: this problem doesn't fix yet, @Fix-Point please create a new patch fix it.

**Pull request merged**